presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -290,6 +290,11 @@ public void addTypes(IdentityHashMap<Expression, Type> types)
         this.types.putAll(types);
     }
 
+    public void replaceType(Expression expression, Type type)
```

Comment:
Could you add a test case that reproduces that? I tried removing `replaceType()` and all the tests passed